### PR TITLE
Notification and Identity client calls

### PR DESCRIPTION
**Pull request by willmichael**

### CODE REVIEW

File: crypto.go [hunk 1]
```diff
@@ -158,6 +158,24 @@ func RandomNonce() Nonce {
 	return &b
 }
 
+// GenerateRandomBytes generate a random number of bytes
+func GenerateRandomBytes(n int) []byte {
+	b := make([]byte, n)
+	_, err := rand.Read(b)
+	if err != nil {
+		// we don't expect this to fail
+		panic("random number generation failed")
```
Comment (efabens) [inline crypto.go]: I don't think we want this to panic

Comment (willmichael) [reply crypto.go]: Normally I would agree, but I'm following the above pattern of using panic when this happens to fail

Comment (lkwdwrd) [reply crypto.go]: I agree, I'd rather return an error than panic here

Comment (willmichael) [reply crypto.go]: hmm some notes in the golang docs refer to 
```
// Read generates len(p) random bytes and writes them into p. It
// always returns len(p) and a nil error.
```
indicating that it always returns a nil error simply to conform to the ioReader interface; however, I will update it in favor of returning error as this note is from the math package not the crypto/rand we're using

**Pull request merged**